arroyo-api/src/connection_tables.rs

```diff
@@ -86,45 +98,235 @@ async fn get_and_validate_connector<E: GenericClient>(
         None
     };
 
-    Ok((connector, connection_id, config, schema))
+    Ok((connector, connection_profile_id, config, schema))
+}
+
+/// Delete a Connection Table
+#[utoipa::path(
+    delete,
+    path = "/v1/connection_tables/{id}",
+    tag = "connection_tables",
+    params(
+        ("id" = String, Path, description = "Connection Table id")
+    ),
+    responses(
+        (status = 200, description = "Deleted connection table"),
+    ),
+)]
+pub(crate) async fn delete_connection_table(
+    State(state): State<AppState>,
+    bearer_auth: BearerAuth,
+    Path(pub_id): Path<String>,
+) -> Result<(), ErrorResp> {
+    let client = client(&state.pool).await?;
+    let auth_data = authenticate(&state.pool, bearer_auth).await?;
+
+    let deleted = api_queries::delete_connection_table()
+        .bind(&client, &auth_data.organization_id, &pub_id)
+        .await
+        .map_err(|e| handle_delete("connection_table", "pipelines", e))?;
+
+    if deleted == 0 {
+        return Err(ErrorResp {
+            status_code: StatusCode::NOT_FOUND,
+            message: "Connection table not found".to_string(),
+        });
+    }
+
+    Ok(())
 }
 
-pub(crate) async fn create(
-    req: CreateConnectionTableReq,
-    auth: AuthData,
-    pool: &Pool,
-) -> Result<(), Status> {
-    let mut c = pool.get().await.map_err(log_and_map)?;
+/// Test a Connection Table
+#[utoipa::path(
+    post,
+    path = "/v1/connection_tables/test",
+    tag = "connection_tables",
+    request_body = ConnectionTablePost,
+    responses(
+        (status = 200, description = "Job output as 'text/event-stream'"),
+    ),
+)]
+pub(crate) async fn test_connection_table(
+    State(state): State<AppState>,
+    bearer_auth: BearerAuth,
+    WithRejection(Json(req), _): WithRejection<Json<ConnectionTablePost>, ApiError>,
+) -> Result<Sse<impl Stream<Item = Result<Event, Infallible>>>, ErrorResp> {
+    let client = client(&state.pool).await?;
+    let auth_data = authenticate(&state.pool, bearer_auth).await?;
+
+    let (connector, _, config, schema) =
+        get_and_validate_connector(&req, &auth_data, &client).await?;
+
+    let (tx, mut rx) = channel(8);
+
+    let (tx2, rx2) = channel(8);
+
+    connector
+        .test(&req.name, &config, &req.config, schema.as_ref(), tx)
+        .map_err(|e| {
+            Status::invalid_argument(format!("Failed to parse config or schema: {:?}", e))
+        })?;
+
+    tokio::spawn(async move {
```

Comment:
instead of having this intermediate task converting from the gRPC-style messages being produced now into SSE messages, it might make sense to update the interface for the connectors — but also fine to leave this for now

Reply:
That would require making connectors depend on axum so that they can construct `axum::response::sse::Event` objects, right?

Reply:
I think that's fine — currently it has a dependency on tonic that won't be needed after these changes

Reply:
Cool, done.